************* Module src.unit_test_generator
unit_test_generator.py:1:0: C0302: Too many lines in module (1809/1000) (too-many-lines)
unit_test_generator.py:56:1: W0511: TODO Import any modules here for whom 'repr' doesn't work, (fixme)
unit_test_generator.py:304:9: W0511: TODO add a "update_types_in_use" method (fixme)
unit_test_generator.py:317:17: W0511: TODO: Fix the root cause of this bug (fixme)
unit_test_generator.py:319:9: W0511: TODO call self.update_types_in_use here (fixme)
unit_test_generator.py:463:21: W0511: TODO try/catch/raise exception (fixme)
unit_test_generator.py:472:21: W0511: TODO try/catch/raise exception (fixme)
unit_test_generator.py:943:5: W0511: TODO remove these assserts and the hash_keys set (fixme)
unit_test_generator.py:949:5: W0511: TODO remove this deepcopy (fixme)
unit_test_generator.py:1015:5: W0511: TODO: Find a better way to do this than a str compare! (fixme)
unit_test_generator.py:1046:9: W0511: TODO: Determine if there's ever more than one leading space (fixme)
unit_test_generator.py:1138:13: W0511: TODO: This is (fixme)
unit_test_generator.py:1273:9: W0511: TODO Fix bug here where function_metadata.coverage_io dictionaries (fixme)
unit_test_generator.py:1279:9: W0511: TODO The below below is likely unnecessary now (fixme)
unit_test_generator.py:1522:5: W0511: TODO: Test this block below (fixme)
unit_test_generator.py:1661:13: W0511: TODO If an arg is a class, construct it (fixme)
unit_test_generator.py:76:0: R0903: Too few public methods (1/2) (too-few-public-methods)
unit_test_generator.py:91:4: W0237: Parameter 'o' has been renamed to 'obj' in overriding 'JsonableEncoder.default' method (arguments-renamed)
unit_test_generator.py:108:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
unit_test_generator.py:135:0: R0902: Too many instance attributes (9/7) (too-many-instance-attributes)
unit_test_generator.py:176:0: R0902: Too many instance attributes (15/7) (too-many-instance-attributes)
unit_test_generator.py:180:4: R0913: Too many arguments (15/5) (too-many-arguments)
unit_test_generator.py:261:35: R1718: Consider using a set comprehension (consider-using-set-comprehension)
unit_test_generator.py:261:39: R1721: Unnecessary use of a comprehension, use list(range(first_source_line_num, last_source_line_num + 1)) instead. (unnecessary-comprehension)
unit_test_generator.py:323:15: E1120: No value for argument 'obj' in function call (no-value-for-parameter)
unit_test_generator.py:401:12: W0602: Using global for 'recursion_depth_per_decoratee' but no assignment is done (global-variable-not-assigned)
unit_test_generator.py:483:19: W0718: Catching too general exception Exception (broad-exception-caught)
unit_test_generator.py:383:8: R0911: Too many return statements (7/6) (too-many-return-statements)
unit_test_generator.py:383:8: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
unit_test_generator.py:526:12: W0719: Raising too general exception: Exception (broad-exception-raised)
unit_test_generator.py:606:0: R0914: Too many local variables (51/15) (too-many-locals)
unit_test_generator.py:606:0: W1113: Keyword argument before variable positional arguments list in the definition of do_the_decorator_thing function (keyword-arg-before-vararg)
unit_test_generator.py:631:4: W0602: Using global for 'all_metadata' but no assignment is done (global-variable-not-assigned)
unit_test_generator.py:631:4: W0602: Using global for 'hashed_inputs' but no assignment is done (global-variable-not-assigned)
unit_test_generator.py:633:4: R1715: Consider using dict.get for getting values from a dict if a key is present or a default if not (consider-using-get)
unit_test_generator.py:693:19: W0718: Catching too general exception Exception (broad-exception-caught)
unit_test_generator.py:732:20: W0123: Use of eval (eval-used)
unit_test_generator.py:791:11: W0718: Catching too general exception Exception (broad-exception-caught)
unit_test_generator.py:824:15: W0718: Catching too general exception Exception (broad-exception-caught)
unit_test_generator.py:606:0: R0912: Too many branches (52/12) (too-many-branches)
unit_test_generator.py:606:0: R0915: Too many statements (209/50) (too-many-statements)
unit_test_generator.py:977:8: W0201: Attribute '_stdout' defined outside __init__ (attribute-defined-outside-init)
unit_test_generator.py:978:21: W0201: Attribute '_stringio' defined outside __init__ (attribute-defined-outside-init)
unit_test_generator.py:1106:0: R0914: Too many local variables (23/15) (too-many-locals)
unit_test_generator.py:1165:4: R1702: Too many nested blocks (6/5) (too-many-nested-blocks)
unit_test_generator.py:1165:4: R1702: Too many nested blocks (6/5) (too-many-nested-blocks)
unit_test_generator.py:1106:0: R0911: Too many return statements (12/6) (too-many-return-statements)
unit_test_generator.py:1106:0: R0912: Too many branches (35/12) (too-many-branches)
unit_test_generator.py:1106:0: R0915: Too many statements (73/50) (too-many-statements)
unit_test_generator.py:1220:12: W0612: Unused variable 'vi' (unused-variable)
unit_test_generator.py:1251:0: R0914: Too many local variables (16/15) (too-many-locals)
unit_test_generator.py:1441:0: R0914: Too many local variables (19/15) (too-many-locals)
unit_test_generator.py:1451:31: R1718: Consider using a set comprehension (consider-using-set-comprehension)
unit_test_generator.py:1451:35: R1721: Unnecessary use of a comprehension, use list(range(first_source_line_num, last_source_line_num + 1)) instead. (unnecessary-comprehension)
unit_test_generator.py:1498:0: R0914: Too many local variables (18/15) (too-many-locals)
unit_test_generator.py:1498:0: R0912: Too many branches (13/12) (too-many-branches)
unit_test_generator.py:1498:0: R0915: Too many statements (54/50) (too-many-statements)
unit_test_generator.py:1580:0: R0913: Too many arguments (6/5) (too-many-arguments)
unit_test_generator.py:1580:0: R0914: Too many local variables (47/15) (too-many-locals)
unit_test_generator.py:1580:0: R0912: Too many branches (34/12) (too-many-branches)
unit_test_generator.py:1580:0: R0915: Too many statements (123/50) (too-many-statements)

------------------------------------------------------------------
Your code has been rated at 9.32/10 (previous run: 9.33/10, -0.01)

